### ISSUE TITLE
macOS home-brew 2.12.1 does not work with layout_python after upgrade

### ISSUE THREAD

**Issue by hostmaster** [direnv/direnv#281]
Hi, 

I've started to get errors from direnv after a recent update. 
```
direnv: loading .envrc
direnv: error exit status 1
```

direnv: v2.12.1 (from home-brew)
zsh: 5.3.1 (x86_64-apple-darwin16.6.0)
macOS: 10.12.5

```
❯ cat .envrc
layout python

TESTENV=test123
```

```
~/Temp/test_python 41s
❯ direnv allow .
direnv: loading .envrc
direnv: error exit status 1

~/Temp/test_python
❯ direnv status
direnv exec path /usr/local/Cellar/direnv/2.12.1/bin/direnv
DIRENV_CONFIG /Users/igor/.config/direnv
DIRENV_BASH /bin/bash
Loaded RC path /Users/igor/Temp/test_python/.envrc
Loaded RC allowed false
Loaded RC allowPath
Found RC path /Users/igor/Temp/test_python/.envrc
Found watch: ".envrc" - 2017-07-03T14:48:51+03:00
Found watch: "../../.config/direnv/allow/1893bee0340fdc09801b07738d0056359e3d7a06ce4d3ad9fea44a9f75bcfbef" - 2017-07-03T14:49:09+03:00
Found RC allowed true
Found RC allowPath /Users/igor/.config/direnv/allow/1893bee0340fdc09801b07738d0056359e3d7a06ce4d3ad9fea44a9f75bcfbef
direnv: loading .envrc
direnv: error exit status 1

```

**Comment by zimbatm** [comment]
hi, what version were you using before the upgrade?

**Comment by hostmaster** [comment]
Most likely it was version 2.11.3. I'm using it right now and it works.

**Comment by hostmaster** [comment]
I downloaded a binary release of `2.12.1` from github and I got the same error. 
```
❯ direnv
direnv v2.12.1
Usage: direnv COMMAND [...ARGS]
```
```
❯ z cloud
direnv: loading .envrc
direnv: error exit status 1
```

**Comment by zimbatm** [comment]
do you mind adding a `set -x` at the top of the .envrc and give me the output?

**Comment by hostmaster** [comment]
```
~/Temp/test_python
direnv: loading .envrc
++ layout python
++ local name=python
++ shift
++ eval layout_python
+++ layout_python
+++ local python=python
+++ shift
direnv: error exit status 1
```

**Comment by hostmaster** [comment]
for 2.11.3 its looks like below 
```
~/Temp/test_python
direnv: loading .envrc
++ layout python
++ local name=python
++ shift
++ eval layout_python
+++ layout_python
+++ local python=python
+++ local old_env=/Users/igor/Temp/test_python/.direnv/virtualenv
+++ unset PYTHONHOME
+++ [[ -d /Users/igor/Temp/test_python/.direnv/virtualenv ]]
+++ local python_version
++++ python -c 'import platform as p;print(p.python_version())'
+++ python_version=2.7.10
+++ [[ -z 2.7.10 ]]
+++ export VIRTUAL_ENV=/Users/igor/Temp/test_python/.direnv/python-2.7.10
+++ VIRTUAL_ENV=/Users/igor/Temp/test_python/.direnv/python-2.7.10
+++ [[ ! -d /Users/igor/Temp/test_python/.direnv/python-2.7.10 ]]
+++ PATH_add /Users/igor/Temp/test_python/.direnv/python-2.7.10/bin
++++ expand_path /Users/igor/Temp/test_python/.direnv/python-2.7.10/bin
++++ /usr/local/bin/direnv.2.11.3 expand_path /Users/igor/Temp/test_python/.direnv/python-2.7.10/bin
+++ PATH=/Users/igor/Temp/test_python/.direnv/python-2.7.10/bin:/Users/igor/.cargo/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/igor/Library/Python/2.7/bin:/usr/local/sbin:/usr/sbin:/sbin:/Users/igor/.cargo/bin:/bin
+++ export PATH
++ TESTENV=test123
+ popd
+ popd
+ /usr/local/bin/direnv.2.11.3 dump
direnv: export +VIRTUAL_ENV ~PATH
```

**Comment by zimbatm** [comment]
Thanks, fixed in 8bb300df95316ae23bdf6c5dc6b903463ff72293

**Comment by zimbatm** [comment]
Released v2.12.2 with the fix
